src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1727,6 +1727,13 @@ void PhaseChaitin::fixup_spills() {
           if( cisc->oper_input_base() > 1 && mach->oper_input_base() <= 1 ) {
             assert( cisc->oper_input_base() == 2, "Only adding one edge");
             cisc->ins_req(1,src);         // Requires a memory edge
+          } else {
+            // In some rare cases:
+            // There is no space reserved for a memory edge before the inputs.
```

Comment:
It sounds like random case but you have very specific case: Mach instructions which uses `stackSlotX`. I think these 2 lines should say that. Something like:
```
// There is no space reserved for a memory edge before the inputs for
// instructions which have "stackSlotX" parameter instead of "memory".
// For example, "MoveF2I_stack_reg".
```

Reply:
@vnkozlov thanks for the suggestion, I put it in.